frontend/src/App.tsx

```diff
@@ -1,64 +1,34 @@
-import { ReactElement, useState, createElement, useEffect } from 'react'
+import { useState, useEffect } from 'react'
 
 import Header from './Header'
 import Sidebar from './Sidebar'
 import Board from './Board'
-import Card from './Card'
 
-class CardState {
-  _id: number  
-  _shape: string | null
-  _color: string | null
-  _fill: string | null
-  _count: number | null
+// https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#handbook-content
+type MapishCardState = {
+  [Property in keyof CardState]: string | number | null
+}
```

Comment:
This is what's called a Mapped Type. It allows us to fill up a type's keys based upon the properties or keys of another type. This makes more sense at the call site down [here](https://github.com/alt-cee/set-finder/pull/2/files#diff-e56cb91573ddb6a97ecd071925fe26504bb5a65f921dc64c63e534162950e1ebR55)